pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTIncrementExpression.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.ast;
+
+/**
+ * Represents an increment or decrement operation on a variable. This represents
+ * both the postfix and prefix forms.
+ *
+ * <p>Since 7.0, this node replaces the nodes PostfixExpression, PreDecrementExpression,
+ * and PreIncrementExpression.
+ *
+ * <p>In prefix form, this has the same precedence as {@linkplain ASTUnaryExpression UnaryExpression}.
+ * In postfix form, this has a precedence greater as {@linkplain ASTUnaryExpression UnaryExpression},
+ * and lower as {@linkplain ASTPrimaryExpression PrimaryExpression}.
+ *
+ * <pre class="grammar">
+ *
+ * IncrementExpr ::= {@linkplain ASTAssignableExpr AssignableExpr} ( "++" | "--" )
+ *                 | ( "++" | "--" ) {@linkplain ASTAssignableExpr AssignableExpr}
+ *
+ * </pre>
+ */
+public final class ASTIncrementExpression extends AbstractJavaExpr implements ASTExpression, LeftRecursiveNode {
```

Comment:
IMO PrefixExpr + PostfixExpr are not so great because they share the two operators `++` and `--`, and I'd rather not duplicate them in separate enums.

AFAIK there's no term for "decrement or increment" expression. Maybe "XcrementExpression"? Just kidding :laughing: 

However there is a term for something that is "either postfix or prefix": *"unary"*. Maybe then, it would make sense to merge IncrementExpression into UnaryExpression and have an attribute for whether it's prefix or postfix instead. That way we have a single node, and also a single enum for unary operators too. Wdyt?

Reply:
Not sure if merging IncrementExpr with UnaryExpr is helpful - In the current use cases in the rules, we would then need to check for the operator, right? (At least in AvoidReassigningLoopVariablesRule)